test/examples/callback_hell.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+var rest = require('restler');
+
+function request(url, cb) {
+    rest.get(url).on('success', function(data, res) {
+        cb(null, data);
+    }).on('error', function(err, res) {
+        cb(err);
+    });
+}
+
+// Callback Hell
+// Change any URL to a non-existant one to get the `err` to fire.
+request('http://google.com', function(err, data1) {
+    if (err) {
+        // return exit(err);
+    } else {
+        
+    }
+    request('http://google.com', function(err, data2) {
+        if (err != undefined)
```

Comment:
Yeah, I was trying to be inclusive for some cases that are technically correct but maybe non-standard, switched them to checking against `null` instead of `undefined`